tests/ClientTest.php

```diff
@@ -0,0 +1,292 @@
+                    "tag": null,
+                    "createdTime": "2021-03-04T21:59:49+00:00"
+                }'
+            ),
+        ]);
+        // Add the history middleware to the handler stack.
+        $requestHistory = [];
+        $history = Middleware::history($requestHistory);
+        $stack = HandlerStack::create($mock);
+        $stack->push($history);
+        $logger = new TestLogger();
+        $client = $this->getClient(['handler' => $stack, 'logger' => $logger, 'userAgent' => 'test agent']);
+        $client->postEvent($this->getPostEventData());
+        /** @var Request $request */
+        $request = $requestHistory[0]['request'];
+        self::assertEquals('test agent', $request->getHeader('User-Agent')[0]);
+        self::assertTrue($logger->hasInfoThatContains('"POST  /1.1" 200 '));
+        self::assertTrue($logger->hasInfoThatContains('test agent'));
+    }
+
+    public function testRetrySuccess(): void
+    {
+        $mock = new MockHandler([
+            new Response(
+                500,
+                ['Content-Type' => 'application/json'],
+                '{"message" => "Out of order"}'
+            ),
+            new Response(
+                501,
+                ['Content-Type' => 'application/json'],
+                'Out of order'
+            ),
+            new Response(
+                200,
+                ['Content-Type' => 'application/json'],
+                '{
+                    "id": "683194249",
+                    "runId": "683194249",
+                    "status": "created",
+                    "desiredStatus": "processing",
+                    "mode": "run",
+                    "component": "keboola.ex-db-snowflake",
+                    "config": "123",
+                    "configRow": null,
+                    "tag": null,
+                    "createdTime": "2021-03-04T21:59:49+00:00"
+                }'
+            ),
+        ]);
+        // Add the history middleware to the handler stack.
+        $requestHistory = [];
+        $history = Middleware::history($requestHistory);
+        $stack = HandlerStack::create($mock);
+        $stack->push($history);
+        $client = $this->getClient(['handler' => $stack]);
+        $client->postEvent($this->getPostEventData());
+        self::assertCount(3, $requestHistory);
+        /** @var Request $request */
+        $request = $requestHistory[0]['request'];
+        self::assertEquals('http://example.com/events/job_failed', $request->getUri()->__toString());
+        $request = $requestHistory[1]['request'];
+        self::assertEquals('http://example.com/events/job_failed', $request->getUri()->__toString());
+        $request = $requestHistory[2]['request'];
+        self::assertEquals('http://example.com/events/job_failed', $request->getUri()->__toString());
+    }
+
+    public function testRetryFailure(): void
+    {
+        $responses = [];
+        for ($i = 0; $i < 30; $i++) {
```

Comment:
Je potreba tu generovat 30 responses? Kdyz mas `'backoffMaxTries' => 1` nemelo by stacit 2? To by se dalo udelat i manualne bez `for`.

Reply:
Stačí dvě, ale ten for se mě tam stejně hodí :)

Reply:
Uz to trochu smrdi programovanim :)